ls1intum/Artemis
# [Bugfix] Start Assessment from Exercise Scores View

### Checklist

### Motivation and Context


Added back functionality to make an assessment for a specific submission.

### Description

- Added new Endpoint to lock a submission for a specific participation
- Added check in exercise scores view to be consistent with other exercise types
- Added check in exercise scores to not display View Assessment button when programming exercise allows no manual result
- New Endpoint is used in resolver for route

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration (you need a programming exercise with participations)
3. Go to Exercise Score view
4. Click on `View Assessment` button, you should see the assessment if it exists or get the lock for the assessment
5. Check how many locks you have -> Assessment Dashboard -> Assessments locked by you (see also Screenshots section)
![image](https://user-images.githubusercontent.com/41108487/100139857-62a3c380-2e90-11eb-8708-35431f2f27a2.png)


### Test Coverage




### Screenshots



Assessments locked by me before trying to assess via exercise score view:
![image](https://user-images.githubusercontent.com/41108487/100136166-f2467380-2e8a-11eb-8c0c-b33c61669599.png)

View Assessment button is now visible all the time and when clicking it and no assessment was yet started you will get the lock: 
![image](https://user-images.githubusercontent.com/41108487/100136314-2f126a80-2e8b-11eb-98b1-2ef9a2b5b859.png)

After clicking on View Assessment button 2 times for not yet startet assessments:
![image](https://user-images.githubusercontent.com/41108487/100136434-5406dd80-2e8b-11eb-831c-15c87670e812.png)
